connect/api/src/main/java/org/apache/kafka/connect/transforms/Transformation.java

```diff
@@ -33,6 +33,9 @@
      * Apply transformation to the {@code record} and return another record object (which may be {@code record} itself) or {@code null},
      * corresponding to a map or filter operation respectively.
      *
+     * A transformation must not mutate the headers of a given {@code record}. If the headers need to be changed
+     * a new record with different headers should be created and returned.
+     *
```

Comment:
Should we also make a note about mutating other things such as the key/value of the record, which is possible for `Struct`, `Map`, and `Array` instances?